### PR TITLE
Lights Out keymap for winry25t

### PR DESCRIPTION
**Pull request by palfrey**
Implements a "Lights Out" mode as per https://en.m.wikipedia.org/wiki/Lights_Out_(game) with a single move having been played on start, and then incrementing by one every time you win.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/winry/winry25tc/keymaps/lightsout/keymap.c [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdlib.h>
+
+#include QMK_KEYBOARD_H
+
+// clang-format off
+enum my_keycodes {
+    K00 = SAFE_RANGE,
```
Comment (drashna) [inline keyboards/winry/winry25tc/keymaps/lightsout/keymap.c:22]:
```suggestion
    K00 = QK_USER,
```

File: keyboards/winry/winry25tc/keymaps/lightsout/keymap.c [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    [0] = LAYOUT(
+        K00, K01, K02, K03, K04,
+        K05, K06, K07, K08, K09,
+        K10, K11, K12, K13, K14,
+        K15, K16, K17, K18, K19,
+        K20, K21, K22, K23, K24
+    )
+};
+
+uint8_t tiles[5][5] = {
```
Comment (drashna) [inline keyboards/winry/winry25tc/keymaps/lightsout/keymap.c:59]: you could probably use bool for this.  Or if you wanted to be fancy, bitmasks

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/winry/winry25tc/keymaps/lightsout/keymap.c (hunk 2 above)
Comment (palfrey) [reply keyboards/winry/winry25tc/keymaps/lightsout/keymap.c]: I've used bools as bitmasks feels like overkill given there's still plenty of space on the device. One for someone else if they want to port it to something smaller :)

**Comment by elpekenin** [comment]
Probably not a great improvement as playing would spam spam keys to the host.... But instead of using custom keycodes and figuring out the position of the key pressed based on manipulations over `keycode`'s value, you could use `record->event.key.col` and `record->event.key.row`, which would work with any keycode. 

That would allow to play on a "real" keymap, by toggling game on/off with a custom keycode (ie: while playing dont send anything to the host)

**Comment by palfrey** [comment]
> Note: Instead of using custom keycodes and figuring out the position of the key pressed based on manipulations over `keycode`'s value, you could use `record->event.key.col` and `record->event.key.row`, which would work with any keycode.
> 
> That would allow to play on a "real" keymap, perhaps having a custom keycode to toggle game on/off (ie: while playing dont send anything to the host)

Oh nice! I've just stripped out my custom setup and re-done it entirely with no-op's (as my only purpose for this particular keyboard is for Lights Out), but yeah, you could do it with a game toggle.

**Comment by palfrey** [comment]
Any thoughts @drashna?

**Review by drashna — APPROVED** [review]

**Pull request merged**